ci/e2e_eks.groovy

```diff
@@ -168,7 +168,7 @@ pipeline {
                 junit testResults: "*.xml", allowEmptyResults: true
             }
         }
-        failure {
+        unsuccessful {
```

Comment:
https://www.jenkins.io/doc/book/pipeline/syntax/

we should use `unsuccessful`, otherwise cleanup steps will not run when the job is timed out.